src/test/test_switches.cpp

```diff
@@ -0,0 +1,315 @@
+/**
+ * This file is part of ExpressLRS
+ * See https://github.com/AlessandroAU/ExpressLRS
+ *
+ * Unit tests for over the air packet encoding, decoding and associated utilities
+ *
+ * Entry point is setup_switches()
+ */
+
+#include <Arduino.h>
+#include <unity.h>
+
+#include "LoRaRadioLib.h"
+#include "CRSF.h"       // has to come after LoraRadioLib.h for R9
+#include "targets.h"
+#include "common.h"
+#include <OTA.h>
+
+// Tests have only been run on TTGO V1, so we have to guard
+// all the testcode to prevent breakages until it's know to do
+// something sensible.
+
+#ifdef TARGET_TTGO_LORA_V1_AS_TX
+
+CRSF crsf;  // need an instance to provide the fields used by the code under test
+
+SX127xDriver Radio; // needed for Radio.TXdataBuffer
```

Comment:
In a future PR I'd like to look at ways to mock this out. Good how it is for now though.

Reply:
At some point we should do an actual API design. The approach of just shoe horning the existing data structures in as params was purely for speed of development (i.e. I was running out of weekend fast).